### ISSUE TITLE
Add rclc_parameter support

### ISSUE THREAD

**Issue by snaens** [xopxe/micro_rosso_platformio#1]
Hey @xopxe ,
do you plan to add ros2 parameters (`rclc_parameters`) to micro_rosso?

Here is an example:
https://micro.ros.org/docs/tutorials/programming_rcl_rclc/parameters/

**Comment by xopxe** [comment]
Yes, that seems to be useful functionality. It's the next thing I will look into.

**Comment by snaens** [comment]
Ok I put some thought into it
It might be smart to save parameters using the arduino `FS.h`
(save them onto the flash)

Then you can save them over reboots and use them as default initializers on the next boot.

This means you could reconfigure micro_rosso with parameters,
so no re-flashing the microcontroller to change I2C pin for instance.

Parameters can also be read-only, for permanent values, like Node name (for when you have multiple microcontrollers and don't want to compile every time).
(you could toggle read-only mode for these with a Pin or something)

**Comment by xopxe** [comment]
I committed an initial version for parameter support ([parameters](https://github.com/xopxe/micro_rosso_platformio/tree/parameters) branch). To make it work you have to provide a colcon.meta file to build micro-ros with more space for services. I'll document that shortly.

Regarding the persistent parameter idea, I've used nvram variables with esp-idf on ESP32, and indeed it is very practical for stuff like sensor calibration data, Wi-Fi credentials, device IDs, and so on. 

To integrate ROS parameters with FS would imply an API to somehow mark variables as persistent, and provide parameter manipulation calls to update the storage behind scene. A "parameter-server" module? Or perhaps it all can be handled from inside the parameter_changed callback, have to investigate this.

**Comment by xopxe** [comment]
Implemented the parameter server and parameter persistence using the Preferences library. For now, it will be in the parameters branch until it gets tested. Parameters are stored and loaded transparently; the user does not have to do anything beyond loading the module. Usage is added to the README.md

The main limitation is that the micro-ros parameter server only supports int64, double, and bool parameters, so no ssids and passwords.
